### PR TITLE
[Merge feature branch to master branch] Implement APIs to support JSON Tabs

### PR DESCRIPTION
**Pull request by sachingupta**
Starting this PR on behalf of interns(adrian, Ashley, Faiz,) working on JSON tabs

### CODE REVIEW

File: src/public/bot.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    query: string;
+    commandId?: string;
+  }
+
+  export interface QueryResponse {
+    attachments: IAttachment[];
+    layout: any;
+    botId: string;
+  }
+  export interface IAttachment {
```
Comment (ydogandjiev) [inline src/public/bot.ts]: NIT: Add empty lines between each interface definitions for better readability

File: src/public/bot.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    // Will add better interfaces later
+    card: any;
+    previewCard: any;
+    previewRawPayload: any;
+    rawPayload: any;
+  }
+  export interface ICommand {
```
Comment (ydogandjiev) [inline src/public/bot.ts]: None of our interfaces have the "I" prefix; consider removing it from this one; i.e. `ICommand` --> `Command`

File: src/public/bot.ts (hunk 1 above)
Comment (ydogandjiev) [inline src/public/bot.ts]: None of our interfaces have the "I" prefix; consider removing it from this one; i.e. `IAttachment ` --> `Attachment `

File: src/public/bot.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    // register handler for callback id
+    GlobalVars.callbacks[messageId] = (success: boolean, response: string | QueryResponse) => {
+      if (success) {
+        onBotQueryResponse(response as QueryResponse);
+      } else {
+        onError(response as string);
+      }
+    };
+  }
+  /**
```
Comment (ydogandjiev) [inline src/public/bot.ts]: NIT: Add empty line between methods for better readability

File: src/public/bot.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    commandId?: string;
+  }
+
+  export interface QueryResponse {
+    attachments: IAttachment[];
+    layout: any;
+    botId: string;
+  }
+  export interface IAttachment {
+    // Will add better interfaces later
```
Comment (ydogandjiev) [inline src/public/bot.ts]: What changes were you planning to make here? Now might be a good time to clean up this interface; e.g. add types and comments.

File: src/public/bot.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+   * @param botRequest query to send to bot.
+   * @param onBotQueryResponse callback to invoke when data is retrieved from bot
+   * @param onError callback to invoke should an error occur
+   */
+  export function sendQuery(
+    botRequest: QueryRequest,
+    onBotQueryResponse?: (data: QueryResponse) => void,
+    onError?: (error: string) => void,
+  ): void {
+    // void for now
```
Comment (ydogandjiev) [inline src/public/bot.ts]: Delete unnecessary comment

Comment (ydogandjiev) [reply src/public/bot.ts]: Same for the other two comments in this method

File: test/MicrosoftTeams.spec.ts [hunk 6]
```diff
@@ -1993,209 +1994,140 @@ describe("MicrosoftTeams", () => {
...
+        response: "Something went wrong..."
       };
-      respondToMessage(message, data.success, data.error);
 
-      // check data is returned properly
-      expect(requestResponse).toBe(false);
-      expect(error).toBe("Something went wrong...");
-    });
-
-    it("should successfully send a request", () => {
```
Comment (sachingupta) [inline test/MicrosoftTeams.spec.ts]: revert this change
